modular/blocksyncer/blocksyncer_options.go

```diff
@@ -1,478 +0,0 @@
-	if err != nil {
-		log.Errorf("failed to GetParserContext err: %v", err)
-		return err
-	}
-	b.parserCtx = ctx
-	b.parserCtx.Indexer = NewIndexer(ctx.EncodingConfig.Marshaler,
-		ctx.Node,
-		ctx.Database,
-		ctx.Modules,
-		b.Name())
-	return nil
-}
-
-// initDB create tables needed by block syncer. It depends on which modules are configured
-func (b *BlockSyncerModular) initDB(useMigrate bool) error {
-
-	var err error
-	for _, module := range b.parserCtx.Modules {
-		if module, ok := module.(modules.PrepareTablesModule); ok {
-			if useMigrate {
-				err = module.AutoMigrate()
-			} else {
-				err = module.PrepareTables()
-			}
-			if err != nil {
-				log.Errorw("failed to PrepareTables/AutoMigrate tables", "error", err)
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-// serve start BlockSyncer rpc service
-func (b *BlockSyncerModular) serve(ctx context.Context) {
-	migrateDBAny := ctx.Value(MigrateDBKey{})
-	if migrateDB, ok := migrateDBAny.(bool); ok && migrateDB {
-		err := b.initDB(true)
-		if err != nil {
-			log.Errorw("fail to init DB", "error", err)
-			return
-		}
-	}
-	// Create a queue that will collect, aggregate, and export blocks and metadata
-	exportQueue := types.NewQueue(100)
-
-	// Create workers
-	worker := parser.NewWorker(b.parserCtx, exportQueue, 0, config.Cfg.Parser.ConcurrentSync)
-	worker.SetIndexer(b.parserCtx.Indexer)
-
-	latestBlockHeight := mustGetLatestHeight(b.parserCtx)
-	Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Store(int64(latestBlockHeight))
-	go b.getLatestBlockHeight(ctx)
-
-	lastDbBlockHeight := uint64(0)
-	for {
-		epoch, err := b.parserCtx.Database.GetEpoch(context.TODO())
-		if err != nil {
-			log.Errorw("failed to get last block height from database", "error", err)
-			continue
-		}
-		lastDbBlockHeight = uint64(epoch.BlockHeight)
-		break
-	}
-
-	// fetch block data
-	go b.quickFetchBlockData(lastDbBlockHeight + 1)
-
-	go b.enqueueNewBlocks(ctx, exportQueue, lastDbBlockHeight+1)
-
-	// Start each blocking worker in a go-routine where the worker consumes jobs
-	go worker.Start(ctx)
-}
-
-// enqueueNewBlocks enqueues new block heights onto the provided queue.
-func (b *BlockSyncerModular) enqueueNewBlocks(context context.Context, exportQueue types.HeightQueue, currHeight uint64) {
-	// Enqueue upcoming heights
-	for {
-		select {
-		case <-context.Done():
-			log.Infof("Receive cancel signal, enqueueNewBlocks routine will stop")
-			// close channel
-			close(exportQueue)
-			return
-		default:
-			{
-				latestBlockHeightAny := Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Load()
-				latestBlockHeight := latestBlockHeightAny.(int64)
-				// Enqueue all heights from the current height up to the latest height
-				for ; currHeight <= uint64(latestBlockHeight); currHeight++ {
-					// log.Debugw("enqueueing new block", "height", currHeight)
-					exportQueue <- currHeight
-				}
-			}
-		}
-	}
-}
-
-func (b *BlockSyncerModular) getLatestBlockHeight(ctx context.Context) {
-	for {
```

Comment:
I think we can use `time.NewTicker()` instead of `sleep` here. right now, if `LatestHeight` returns  error, it will keep doing this without any wait, thus cause high CPU as well.